### PR TITLE
[cake] add cake pool subgraph

### PR DESCRIPTION
**Pull request by chef-jojo**
Changes proposed in this pull request:
- Add cake pool subgraph to add the syrup pools voting power back
- For proposal snapshot < 16300686, add missing master chef and pool balance.

### CODE REVIEW

**Review by ChaituVR — CHANGES_REQUESTED** [review]
Hi @chef-jojo This strategy fails when we pass more addresses (500 addresses)
Try running 
```
npm test --strategy=cake --more=500
```
and also these changes increase the number of requests being sent from a strategy, the requests should not be more than 5, 
https://github.com/snapshot-labs/snapshot-strategies#checklist-for-adding-a-new-strategy

**Comment by chef-jojo** [comment]
@ChaituVR, merged some calls! We have a if condition in block number
https://github.com/snapshot-labs/snapshot-strategies/blob/91d2cbeb03b914f47180086beeba861828d9bb14/src/strategies/cake/index.ts#L140-L143

snapshot < onChainVPBlockNumber: 5 calls
snapshot >= onChainVPBlockNumber: 3 calls

**Review by ChaituVR — CHANGES_REQUESTED** [review]

File: src/strategies/cake/index.ts [hunk 1]
```diff
@@ -66,4 +181,91 @@ export async function strategy(
...
+    addresses,
+    {
+      chefAddress: MASTER_CHEF_ADDRESS.v1,
+      uniPairAddress: CAKE_BNB_LP_ADDRESS,
+      pid: '251',
+      symbol: 'CAKE-BNB LP',
+      tokenIndex: 0
+    },
+    snapshot
+  );
```
[the remark this answers is not in the document]
Comment (ChaituVR) [reply src/strategies/cake/index.ts:198]: oh, this is just for else case right. hmmm

[the remark this answers is not in the document]
Comment (ChaituVR) [reply src/strategies/cake/index.ts:198]: you can ignore this comment then :)

**Review by ChaituVR — APPROVED** [review]

**Pull request merged**